src/Symfony/Bridge/Twig/Extension/FormExtension.php

```diff
@@ -79,6 +96,83 @@ public function getTests(): array
             new TwigTest('rootform', 'Symfony\Bridge\Twig\Extension\twig_is_root_form'),
         ];
     }
+
+    public function getFieldName(FormView $view): string
+    {
+        $view->setRendered();
+
+        return $view->vars['full_name'];
+    }
+
+    public function getFieldValue(FormView $view): string
+    {
+        return $view->vars['value'];
+    }
+
+    public function getFieldLabel(FormView $view): string
+    {
+        return $this->createFieldTranslation(
+            $view->vars['label'],
+            $view->vars['label_translation_parameters'] ?: [],
+            $view->vars['translation_domain']
+        );
+    }
+
+    public function getFieldHelp(FormView $view): string
+    {
+        return $this->createFieldTranslation(
+            $view->vars['help'],
+            $view->vars['help_translation_parameters'] ?: [],
+            $view->vars['translation_domain']
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getFieldErrors(FormView $view): iterable
+    {
+        /** @var FormError $error */
+        foreach ($view->vars['errors'] as $error) {
+            yield $error->getMessage();
+        }
+    }
+
+    /**
+     * @return Translatable[]|Translatable[][]
+     */
+    public function getFieldChoices(FormView $view): iterable
+    {
+        yield from $this->createFieldChoicesList(
+            $view->vars['choices'],
+            $view->vars['choice_translation_domain'] ?: 'messages'
+        );
+    }
+
+    private function createFieldChoicesList(iterable $choices, string $translationDomain): iterable
+    {
+        foreach ($choices as $choice) {
+            $translatableLabel = $this->createFieldTranslation($choice->label, [], $translationDomain);
+
+            if ($choice instanceof ChoiceGroupView) {
+                yield $translatableLabel => $this->createFieldChoicesList($choice, $translationDomain);
+
+                continue;
+            }
+
+            /* @var ChoiceView $choice */
+            yield $translatableLabel => $choice->value;
+        }
+    }
+
+    private function createFieldTranslation(?string $value, array $parameters, ?string $domain): string
+    {
+        if (!$this->translator || !$value) {
+            return $value;
+        }
+
+        return $this->translator->trans($value, $parameters, $domain ?: 'messages');
```

Comment:
I would also remove the default domain from here (i.e. `?: 'messages'`) it's an implementation detail already covered by the translator. And if someone has a custom translator implementation it's their responsibility to determine the default domain if null.
https://github.com/symfony/symfony/blob/2a5710f6419639b1f8396d81d76452a65b07e939/src/Symfony/Component/Translation/Translator.php#L202-L204